Mathlib/RingTheory/Polynomial/Chebyshev.lean

```diff
@@ -78,26 +99,55 @@ theorem T_zero : T R 0 = 1 := rfl
 theorem T_one : T R 1 = X := rfl
 #align polynomial.chebyshev.T_one Polynomial.Chebyshev.T_one
 
-@[simp]
-theorem T_add_two (n : ℕ) : T R (n + 2) = 2 * X * T R (n + 1) - T R n := by rw [T]
-#align polynomial.chebyshev.T_add_two Polynomial.Chebyshev.T_add_two
+theorem T_neg_one : T R (-1) = X := (by ring : 2 * X * 1 - X = X)
 
-theorem T_two : T R 2 = 2 * X ^ 2 - 1 := by simp only [T, sub_left_inj, sq, mul_assoc]
+theorem T_two : T R 2 = 2 * X ^ 2 - 1 := by
+  simpa [pow_two, mul_assoc] using T_add_two R 0
 #align polynomial.chebyshev.T_two Polynomial.Chebyshev.T_two
 
-theorem T_of_two_le (n : ℕ) (h : 2 ≤ n) : T R n = 2 * X * T R (n - 1) - T R (n - 2) := by
-  obtain ⟨n, rfl⟩ := Nat.exists_eq_add_of_le h
-  rw [add_comm]
-  exact T_add_two R n
-#align polynomial.chebyshev.T_of_two_le Polynomial.Chebyshev.T_of_two_le
-
-/-- `U n` is the `n`-th Chebyshev polynomial of the second kind -/
-noncomputable def U : ℕ → R[X]
+@[simp]
+theorem T_neg (n : ℤ) : T R (-n) = T R n := by
+  induction n using T.induct R with
```

Comment:
Possibly worth defining a custom induction principle to avoid dealing with `Int.negSucc` etc:
```
@[elab_as_elim]
theorem T.induct' (motive : ℤ → Prop)
  (zero : motive 0)
  (one : motive 1)
  (add_two : ∀ (n : ℕ), motive (↑n + 1) → motive ↑n → motive (↑n + 2))
  (neg_add_one : ∀ (n : ℕ), motive (-↑n) → motive (-↑n + 1) → motive (-↑n - 1)) :
  ∀ (a : ℤ), motive a :=
  T.induct R motive zero one add_two fun n hn hnm => by
    simpa only [Int.negSucc_eq, neg_add] using neg_add_one n hn hnm
```